### ISSUE TITLE
Older journal records prevent XML upload for newer journals

### ISSUE THREAD

**Issue by dommitchell** [DOAJ/doaj#679]
I need some feedback ASAP. 

Publisher account 22130896 is unable to upload XML for two journals:

https://doaj.org/admin/journals?source={%22query%22:{%22query_string%22:{%22query%22:%220550-3213%22,%22default_operator%22:%22AND%22}}}

https://doaj.org/admin/journals?source={%22query%22%3A{%22query_string%22%3A{%22query%22%3A%220370\\-2693%22%2C%22default_field%22%3A%22index.issn.exact%22%2C%22default_operator%22%3A%22AND%22}}%2C%22from%22%3A0%2C%22size%22%3A10}

![image001](https://cloud.githubusercontent.com/assets/5955396/7247092/a4f40d4e-e805-11e4-81b6-d4d72fd07206.png)

The XML is mass generated and other XML uploads for this publisher work without problem.

You can see that both journals have older, rejected entries in the database and I wonder if this could be causing the XML failure. I have been through the XML and exhausted all my XML coding knowledge of why the XML might be failing so I'd like your thoughts on my theory.

**Comment by emanuil-tolev** [comment]
Fwd / dropbox the files (I assume you have them if you've investigated them yourself)?

**Comment by dommitchell** [comment]
Forwarded.

**Comment by richard-jones** [comment]
This doesn't look like an xml error to me.

Import failure at this level is caused by issns in the supplied articles being in the following situations:

1. Not matching any journal that is in the index (in_doaj=True or False)
2. Matching a journal that does not have an owner (this should be rare to non-existent, and would indicate a data error on the doaj-side part most lilkely)
3. Matching multiple journals with different owners (again, rare, and indicates a data error doaj-side)
4. Matching a journal with an owner different to the user account being uploaded

Therefore, to import, the article must contain issn(s) that match exactly one journal that is currently in_doaj (withdrawn journals will not be matched), and that journal must be controlled by the uploading user account.  If there is more than one issn on the article, between them they must meet this criteria.

First thing I'd do, then, is check that the issns are for the right user account.

If there are multiple copies of the journal with the same issn (e.g. one withdrawn and one not) this shouldn't matter provided that they are both owned by the same user account.  If they are not owned by the same user, this /would/ cause this error.

**Comment by dommitchell** [comment]
> First thing I'd do, then, is check that the issns are for the right user account.

Yup, done that already and can confirm that every article in both XML files has the correct ISSNs and that they match the two journal entries in DOAJ.

Have also checked your numbers 1 to 4 above and all is correct

**Comment by richard-jones** [comment]
most odd.  Ok, will require further investigation - will look into it once we've got the roll-out sorted.

**Comment by dommitchell** [comment]
#426 might be the same problem

**Comment by emanuil-tolev** [comment]
leaving to @richard-jones 

here's some info for Richard that might be helpful in diagnosing some of those if the problem doesn't end with what we discussed in meeting:

Missing an owner and in DOAJ (4): https://doajgate.cottagelabs.com/doaj/journal/_search?q=_missing_:admin.owner%20AND%20admin.in_doaj:true

Missing an owner and NOT in DOAJ (1675): https://doajgate.cottagelabs.com/doaj/journal/_search?q=_missing_:admin.owner%20AND%20admin.in_doaj:false

Missing an owner and in_doaj information (0): https://doajgate.cottagelabs.com/doaj/journal/_search?q=_missing_:admin.owner%20AND%20_missing_:admin.in_doaj

**Comment by dommitchell** [comment]
Updated 2nd URL to show the 2nd journal (NUPHB in the screenshot)

**Comment by richard-jones** [comment]
Ok, this /is/ the same problem as #426 but expressed in a slightly different way.

The Journal model defines a property "owner" which returns an empty string if the owner is not set:

https://github.com/DOAJ/doaj/blob/develop/portality/models/journal.py#L434

In this case, the two expressions of the journal (one in_doaj the other withdrawn), differ in that one has an owner and the other does not (i.e. it does not have a different owner, it just doesn't have one).

So when the code which checks for legitimacy of ownership looks at these two journals and asks for their owners, the withdrawn one responds with an empty string.  This means that the None check here does not catch it:

https://github.com/DOAJ/doaj/blob/develop/portality/article.py#L75

So, the list of possible owners becomes ['theowner', ''] rather than ['theowner'].  This means that when we ask if "theowner" is the unambiguous owner of this journal, the answer is no.

The interesting thing is that I think this behaviour is correct, although the code that does it should be more explicit.  The ownership of this ISSN /is/ confused, just in a different way to #426.  If we allowed upload to this owner, and then a different owner were applied to the withdrawn item, we would be back in the same situation as where the same issn is owned by different users, and we don't permit article upload in this case.

@dommitchell - for you guys, I think there's a decision to make about what the behaviours here should actually be.  The options we discussed are:

1. Treat this as an admin task when it happens - DOAJ admins need to assign the correct owners even to withdrawn journals, or delete the withdrawn journals altogether.

2. Just let it go - we could relax the constraint on upload, so that it only considers journals in doaj.  This could give rise to confusions in the future, and there is a risk that articles will appear next to journals that the uploader of the article doesn't own.  My guess is the risk is small, but it is there.

3. some as yet undefined clever behaviour - we could assign ownership automatically to the withdrawn journal if no owner is set, we could look to make enhanced error messages, and perhaps notifications to admins that something has happened.  Or we could have controls on the withdraw/approve process which would prevent you putting a journal into the doaj if there is a duplicate issn there, allowing us to go for option (2) above.
